### DISCUSSION TITLE
Trying to add IKEA Bulbs LED2002G5/LED1935C3 according to tutorial - Err: cannot find module 'zigbee-...' #341

### DISCUSSION THREAD

**Discussion by stodla123** [Koenkk/zigbee2mqtt#10411] [General]
Hello,

(i also posted this here: https://github.com/itavero/homebridge-z2m/issues/341)

i am trying to add IKEA Bulb LED2002G5 (TRADFRIbulbE14WSglobeopal470lm) and LED1935C3 (TRADFRIbulbE14WWclear250lm)
to Zigbee2MQTT.

The problem happens with V1.6.2 and V1.7.0

After pairing to my existing network (including other IKEA bulbs which work) i get this messages:

Dec 28 15:56:39 homebridge npm[430]: Zigbee2MQTT:info 2021-12-28 15:56:39: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0x804b50fffece16a6"},"type":"device_announced"}'

Dec 28 15:57:47 homebridge npm[430]: Zigbee2MQTT:warn 2021-12-28 15:57:47: Received message from unsupported device with Zigbee model 'TRADFRIbulbE14WSglobeopal470lm' and manufacturer name 'IKEA of Sweden'

Dec 28 15:57:47 homebridge npm[430]: Zigbee2MQTT:warn 2021-12-28 15:57:47: Please see: https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html.

After carefully following the steps it looks this way as described in the tutorial above:

pi@homebridge:/opt/zigbee2mqtt/data $ ls -all
total 52
drwxr-xr-x 3 pi pi 4096 Dec 28 16:04 .
drwxr-xr-x 11 pi pi 4096 Feb 19 2021 ..
-rwxr-xr-x 1 pi pi 1340 Dec 28 16:04 TRADFRIbulbE14WSglobeopal470lm.js
-rwxrwxrwx 1 pi pi 1340 Nov 20 09:32 TRADFRIbulbE14WWclear250lm.js
-rw-r--r-- 1 pi pi 1167 Dec 25 17:11 configuration.yaml
-rw-r--r-- 1 pi pi 22595 Dec 28 15:06 database.db
drwxr-xr-x 12 pi pi 4096 Dec 26 17:05 log
-rw-r--r-- 1 pi pi 3116 Dec 28 16:00 state.json

This is my configuration.yaml:

homeassistant: false
permit_join: true
mqtt:
base_topic: zigbee2mqtt
server: 'mqtt://localhost:1883'
serial:
port: /dev/ttyACM0
devices:
'0x00124b0022670e7a':
friendly_name: '0x00124b0022670e7a'
'0x0017880109164423':

.
.
.

'0x8cf681fffe46edd9':
friendly_name: '0x8cf681fffe46edd9'
'0x001788010b0967d7':
friendly_name: '0x001788010b0967d7'
advanced:
log_level: debug
external_converters:

TRADFRIbulbE14WSglobeopal470lm.js
TRADFRIbulbE14WWclear250lm.js
Then i get the following error:

pi@homebridge:/opt/zigbee2mqtt/data $ sudo journalctl -u zigbee2mqtt.service -f
-- Logs begin at Sun 2021-12-26 16:17:02 CET. --
Dec 28 16:06:39 homebridge npm[430]: Zigbee2MQTT:info 2021-12-28 16:06:39: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Update available for '0x001788010b0967d7'","meta":{"device":"0x001788010b0967d7","status":"available"},"type":"ota_update"}'
Dec 28 16:07:22 homebridge systemd[1]: Stopping zigbee2mqtt...
Dec 28 16:07:22 homebridge npm[430]: Zigbee2MQTT:info 2021-12-28 16:07:22: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload 'offline'
Dec 28 16:07:22 homebridge npm[430]: Zigbee2MQTT:info 2021-12-28 16:07:22: Disconnecting from MQTT server
Dec 28 16:07:24 homebridge npm[430]: Zigbee2MQTT:info 2021-12-28 16:07:24: zigbee-herdsman stopped
Dec 28 16:07:24 homebridge systemd[1]: zigbee2mqtt.service: Succeeded.
Dec 28 16:07:24 homebridge systemd[1]: Stopped zigbee2mqtt.
Dec 28 16:10:02 homebridge systemd[1]: Started zigbee2mqtt.
Dec 28 16:10:14 homebridge npm[12944]: > zigbee2mqtt@1.17.1 start /opt/zigbee2mqtt
Dec 28 16:10:14 homebridge npm[12944]: > node index.js
Dec 28 16:10:45 homebridge npm[12944]: internal/modules/cjs/loader.js:905
Dec 28 16:10:45 homebridge npm[12944]: throw err;
Dec 28 16:10:45 homebridge npm[12944]: ^
Dec 28 16:10:45 homebridge npm[12944]: Error: Cannot find module 'zigbee-herdsman-converters/lib/extend'
Dec 28 16:10:45 homebridge npm[12944]: Require stack:
Dec 28 16:10:45 homebridge npm[12944]: - /opt/zigbee2mqtt/lib/util/utils.js
Dec 28 16:10:45 homebridge npm[12944]: - /opt/zigbee2mqtt/lib/util/settings.js
Dec 28 16:10:45 homebridge npm[12944]: - /opt/zigbee2mqtt/index.js
Dec 28 16:10:45 homebridge npm[12944]: at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
Dec 28 16:10:45 homebridge npm[12944]: at Function.Module._load (internal/modules/cjs/loader.js:746:27)
Dec 28 16:10:45 homebridge npm[12944]: at Module.require (internal/modules/cjs/loader.js:974:19)
Dec 28 16:10:45 homebridge npm[12944]: at require (internal/modules/cjs/helpers.js:93:18)
Dec 28 16:10:45 homebridge npm[12944]: at evalmachine.:5:16
Dec 28 16:10:45 homebridge npm[12944]: at Script.runInContext (vm.js:144:12)
Dec 28 16:10:45 homebridge npm[12944]: at Script.runInNewContext (vm.js:149:17)
Dec 28 16:10:45 homebridge npm[12944]: at Object.runInNewContext (vm.js:303:38)
Dec 28 16:10:45 homebridge npm[12944]: at Object.getExternalConvertersDefinitions (/opt/zigbee2mqtt/lib/util/utils.js:159:16)
Dec 28 16:10:45 homebridge npm[12944]: at getExternalConvertersDefinitions.next () {
Dec 28 16:10:45 homebridge npm[12944]: code: 'MODULE_NOT_FOUND',
Dec 28 16:10:45 homebridge npm[12944]: requireStack: [
Dec 28 16:10:45 homebridge npm[12944]: '/opt/zigbee2mqtt/lib/util/utils.js',
Dec 28 16:10:45 homebridge npm[12944]: '/opt/zigbee2mqtt/lib/util/settings.js',
Dec 28 16:10:45 homebridge npm[12944]: '/opt/zigbee2mqtt/index.js'
Dec 28 16:10:45 homebridge npm[12944]: ]
Dec 28 16:10:45 homebridge npm[12944]: }
Dec 28 16:10:45 homebridge npm[12944]: npm ERR! code ELIFECYCLE
Dec 28 16:10:45 homebridge npm[12944]: npm ERR! errno 1
Dec 28 16:10:46 homebridge npm[12944]: npm ERR! zigbee2mqtt@1.17.1 start: node index.js
Dec 28 16:10:46 homebridge npm[12944]: npm ERR! Exit status 1
Dec 28 16:10:46 homebridge npm[12944]: npm ERR!
Dec 28 16:10:46 homebridge npm[12944]: npm ERR! Failed at the zigbee2mqtt@1.17.1 start script.
Dec 28 16:10:46 homebridge npm[12944]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
Dec 28 16:10:46 homebridge npm[12944]: npm ERR! A complete log of this run can be found in:
Dec 28 16:10:46 homebridge npm[12944]: npm ERR! /home/pi/.npm/_logs/2021-12-28T15_10_46_137Z-debug.log
Dec 28 16:10:46 homebridge systemd[1]: zigbee2mqtt.service: Main process exited, code=exited, status=1/FAILURE
Dec 28 16:10:46 homebridge systemd[1]: zigbee2mqtt.service: Failed with result 'exit-code'.
Dec 28 16:10:46 homebridge systemd[1]: zigbee2mqtt.service: Service RestartSec=100ms expired, scheduling restart.
Dec 28 16:10:46 homebridge systemd[1]: zigbee2mqtt.service: Scheduled restart job, restart counter is at 1.
Dec 28 16:10:46 homebridge systemd[1]: Stopped zigbee2mqtt.

some details:

pi@homebridge:/opt/zigbee2mqtt/data $ npm -v
6.14.15

pi@homebridge:/opt/zigbee2mqtt/data $ lsb_release -a
No LSB modules are available.
Distributor ID: Raspbian
Description: Raspbian GNU/Linux 10 (buster)
Release: 10
Codename: buster

pi@homebridge:/opt/zigbee2mqtt/data $ cat /etc/issue
Homebridge Raspbian - \n (\r)

Node.js Version v14.18.0
Npm Version v6.14.15

Any help appreciated thank you!!

**Comment by reefland** [comment]
I was just doing something like this, in the `configuration.yaml`, I had to format it more like this:
```
external_converters:
  - TRADFRIbulbE14WSglobeopal470lm.js
  - TRADFRIbulbE14WWclear250lm.js
```

I didn't see the spaces and dash in your example.

**Comment by itavero** [comment]
Perhaps add some formatting to your message to make it more readable.

Also, which version of Zigbee2MQTT are you running?
As far as I can tell these devices seem to be already supported (if I look at the [Supported devices](https://www.zigbee2mqtt.io/devices/LED1935C3.html) on the Zigbee2MQTT website.

**Comment by stodla123** [comment]
Thank you for the input, but it has been resolved i was running an outdatet version of z2m, my bad.
